packages/tools/kadena-cli/src/account/utils/accountHelpers.ts

```diff
@@ -0,0 +1,52 @@
+import yaml from 'js-yaml';
+import { readdirSync } from 'node:fs';
+import { join } from 'node:path';
+import { z } from 'zod';
+
+import { ACCOUNT_DIR } from '../../constants/config.js';
+import { services } from '../../services/index.js';
+import { isEmpty } from '../utils/addHelpers.js';
+
+interface IAccount {
+  name: string;
+  alias: string;
+}
+
+const accountAliasFileSchema = z.object({
+  name: z.string(),
+  fungible: z.string(),
+  publicKeys: z.array(z.string()),
+  predicate: z.string(),
+});
+
+const readAccountFromFile = async (accountFile: string): Promise<IAccount> => {
+  const accountAlias = accountFile.split('.')[0];
+  const content = await services.filesystem.readFile(
+    join(ACCOUNT_DIR, accountFile),
+  );
+  const account = content !== null ? yaml.load(content) : null;
+  const parsedContent = accountAliasFileSchema.parse(account);
```

Comment:
Make sure zod `.parse()` is wrapped in a try{}catch(){} that gracefully handles when it errors with a good error message.

Reply:
or there is `.safeParse()` if that makes it easier